frontend/src/app/core/business.service.ts

```diff
@@ -48,107 +37,286 @@ export class BusinessService {
   }
 
   /**
-   *
+   * Registers a new user profile
    * @param {User} user
-   * @returns {Observable<IRegisterResponse>}
+   * @returns {Observable<User>}
    */
-  register(user: User): Observable<IRegisterResponse> {
-    return this.resourceService.register(AuthFactory.createRegisterRequest(user));
+  register(user: User): Observable<User> {
+    return Observable.create( (observer) => {
+      this.resourceService.register(AuthFactory.createRegisterRequest(user))
+      .subscribe(res => {
+        observer.next(User.fromJSON(res.user));
+      }, err => {
+        const response: IBaseResponse = err.error;
+        console.log('register failed, ', response.message);
+        return observer.error(response.message);
```

Comment:
die 3 Zeilen sind über identisch - könnte man evtl. in eine eigene Methode auslagern.
Das 'return' beim observer.error -> ich weiss nicht so genau - brauchts das?

Reply:
In eigene Funktion ausgelagert.